src/mpid/ch4/shm/xpmem/xpmem_init.c

```diff
@@ -67,6 +62,38 @@ int MPIDI_XPMEM_mpi_init_hook(int rank, int size, int *n_vcis_provided, int *tag
     /* Initialize other global parameters */
     MPIDI_XPMEM_global.sys_page_sz = (size_t) sysconf(_SC_PAGESIZE);
 
+    /* Initialize coop counter */
+    MPIDI_XPMEM_global.num_pending_cnt = 0;
```

Comment:
Maybe it is better to briefly describe how the counter is shared (i.e., direct portion is attached at init and indirect portion is dynamically attached at p2p) and the expected performance impact (e.g., indirect portion pays only one-time attach overhead and can be reused by later messages) here.